test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -282,6 +284,40 @@ def test_emacs(image):
     _run_cmd_on_container(container_name, ctx, "emacs -version")
 
 
+@pytest.mark.model("N/A")
+@pytest.mark.integration("sagemaker python sdk")
+def test_sm_pysdk_version(image):
+    """
+    Simply verify that we have sagemaker > 2 in the python sdk
+
+    :param image: ECR image URI
+    """
+    ctx = Context()
+    container_name = _get_container_name("sm_pysdk", image)
+    _start_container(container_name, image, ctx)
+
+    framework, framework_version = get_framework_and_version_from_tag(image)
+
+    if framework == "tensorflow":
+        if is_tf_version("2", image):
+            framework = "tensorflow2"
+        elif is_tf_version("1", image):
+            framework = "tensorflow1"
+
+    sm_pysdk_v2 = {"tensorflow2": "2.3.0", "tensorflow1": "1.15.0"}
+
+    # Ensure
+    sm_version = _run_cmd_on_container(
+        container_name, ctx, "python -c 'import sagemaker; print(sagemaker.__version__)", warn=True).stdout.strip()
+
+    if "NameError" in sm_version:
+        LOGGER.warn("This container does not have SageMaker python sdk")
+    else:
+        if version.parse(framework_version) > version.parse(sm_pysdk_v2[framework]):
```

Comment:
This check should apply to all TF versions, because we release all of them with `sagemaker<2` installed.

Also, for a strict lower-bound comparison, you can use the `is_below_tf_version` function